src/index.js

```diff
@@ -0,0 +1,232 @@
+    // console.log(equat);
+    checkFirstParam(equat[0]);
+    for (let i = 0; i < equat.length; i++) {
+        let equat1 = equat[i];
+        console.log("nachalo: '" + equat1 + '\'');
+        // console.log("equat1 " + equat1 + " , i: " + i );
+        signValid(equat1, equat[i - 1], equat[i + 1]);
+        if (equat1 === '=') {
+            equation.flag = 1;
+        } else if (equat1.indexOf('X', 0) === 0 && (equat[i - 1] !== "*")) {
+            console.log("poisk X: " + equat1);
+            if (equat1 === 'X') {
+                temp = "X^1";
+            } else if (equat1.indexOf('X^', 0) === 0) {
+                temp = "X^" + equat1[2];
+            }
+            console.log("temp: " + temp);
+            getPow(temp, equat[i - 1], 1);
+        } else if (equat1 === '*' && i <= equat.length - 1 && i !== 0) {
+            console.log("1) equat1: " + equat1);
+            koef = numberValid(equat[i - 1]);
+            if (equat[i + 1].indexOf('^', 0) === -1 && equat[i + 1].indexOf('X', 0) === 0) {
+                equat[i + 1] += "^1";
+                console.log("equat i + 1: " + equat[i + 1]);
+                getPow(equat[i + 1], equat[i - 2], koef);
+            } else {
+                if (i <= equat.length - 2 && getPow(equat[i + 1], equat[i - 2], koef) < 0) {
+                    console.log("here2");
+                    return errors(2, equation.pow);
+                }
+            }
+        } else if (i <= equat.length - 1 && equat[i + 1 ] !== '*' && equat1.indexOf('X^', 0) === -1
+            && !isSign(equat1) && numberValid(equat1) !== -1) {
+            console.log("2) equat1: " + equat1);
+            console.log(equation);
+            if (i === 0)
+                getPow('X^0', '+', equat1);
+            else
+                getPow('X^0', equat[i - 1], equat1);
+        }
+    }
+    return (1);
+
+}
+
+function checkFirstParam(equat1) {
+    // console.log(equat1);
+    if (equat1.includes('*') || equat1.includes('+')) {
+        return (errors(1));
+    }
+    return (1);
+}
+
+function getPow(equat1, sign, koef) {
+    koef = +koef;
+    // console.log('\n\nequat: ' + equat1);
+    let pow = equat1.slice(equat1.indexOf('^') + 1, equat1.length);
+
+    if (!equation.pows[pow]) {
+        equation.pows[pow] = 0;
+    }
+    if (sign === '-') {
+        koef = -koef;
+    }
+    (equation.flag === 0) ? equation.pows[pow] += koef : equation.pows[pow] -= koef;
+    // console.log(equation);
+    return (1);
+}
+
+function reducedForm() {
+    let res = "Reduced Form: ";
+    console.log(equation);
+    Object.keys(equation.pows).map((key, index) => {
+        const sign = equation.pows[key] >= 0 ? '+' : '-';
+        if (index === 0 && sign === '+') {
+            res += `${equation.pows[key]} * X^${key} `
+        } else {
+            res += `${sign} ${Math.abs(equation.pows[key])} * X^${key} `
+        }
+    });
+    res += "= 0";
+    console.log(res);
+    return (checkPows());
+}
+
+function checkPows() {
```

Comment:
name function getMaxPow and return from here max pow
console.log in main what u want